articles/Getting started with structs sorting in C++/index.md

```diff
@@ -0,0 +1,266 @@
+{
+	string name;
+	int age;
+}Employee,Student;
+
+int main()
+{
+	Employee.name = "Frank";
+	Employee.age = 63;
+	
+	Student.name = "Clare";
+	Student.age = 53;
+}
+```
+### Accessing struct elements
+The struct elements can also be easily accessed by still using the dot (.) as illustrated below.
+```c++
+#include <bits/stdc++.h>
+using namespace std;
+
+struct Person
+{
+	string name;
+	int age;
+}Employee,Student;
+
+int main()
+{
+	Employee.name = "Frank";
+	Employee.age = 63;
+	
+	Student.name = "Clare";
+	Student.age = 53;
+	
+	// we access the elements here using the "." and print them 
+	cout << Employee.name <<" "<< Employee.age << endl ;
+	cout << Student.name << " "<< Student.age;
+}
+```
+Output:
+```
+Frank 63
+Clare 53
+
+```
+
+### working with arrays of a structure
+It is an array consisting of structs. It can be used to store many instances of a struct object. We are going to create an array of structs that shall use for loops for the collection of the user input and also displaying it.
+```c++
+#include <iostream>
+using namespace std;
+
+typedef struct Person
+{
+	string name;
+	int age;
+}Student;
+
+int main()
+{
+	int i,n;
+	cout<< "Enter number of students :";
+	cin>> n;
+	
+	// we set the array to the number of students according to the users input
+	Student s_array[n];
+	
+	// we prompt the user to enter input 
+	cout<<"Enter Name of Student followed by age \n\n";
+	// the for loop below collects the user input for the specified number of students
+	for(i=0;i<n;i++){
+		cout<<"________\n"; // just a line to enhance readability
+		cin>>s_array[i].name;
+		cin>>s_array[i].age;
+	}
+	
+	cout<<"\nHere is the students list:\n";
+	//the for loop below prints the user output in the oder of entry
+	for(i=0;i<n;i++){
+		cout<<s_array[i].name <<"   "<<s_array[i].age <<endl;
+		
+	}
+}
+```
+###  Sorting Structs
+Now that we are familiar with how Structs work, let us get started with the sorting. We are going to add more attributes(members) of a person to the `Person` Struct to make it much more practical. In addition to that, we are also going to add a comparator function which will be used by the STL sort function for sorting in our Structs. In our comparator function we decide which member we want to use to sort the arrays of structs. In our case, we are going to sort them using their age, from the youngest to the oldest by using the less-than operator (<) .ie:
+```c++
+bool compare( Student a, Student b){
+	if(a.age < b.age)
+		return 1;
+	else 
+		return 0;
+	
+}
+```
+To sort them from the oldest to the youngest, you should just go ahead and replace the '<' with the '>' operator. If you also want to change the sorting member, you just need to replace it where `age` is mentioned in the comparator function. The fun fact is, you may also sort them alphabetically by modifying the comparator function to look like this:
+```c++
+bool compare( Student a, Student b){
+	if(a.name < b.name)
+		return 1;
+	else 
+		return 0;
+	
+}
+```
```

Comment:
I use it to tell the sort() function which member of the struct to use for sorting eg 1:
this tells the sort() to sort the structs according to their age.
```c++
bool compare( Student a, Student b){
	if(a.age < b.age)
		return 1;
	else 
		return 0;
	
}
```

eg 2:
This one uses roll_no for sorting instead of age
```c++
bool compare( Student a, Student b){
	if(a.roll_no < b.roll_no)
		return 1;
	else 
		return 0;
	
}
```
It simply for giving the programmer more control of the sort()